FXyz-Samples/src/main/java/org/fxyz3d/samples/utilities/RayShooting.java

```diff
@@ -74,33 +77,32 @@ public class RayShooting extends ShapeBaseSample {
     
     private final AmbientLight rayLight = new AmbientLight();
     protected PointLight light, light2, light3;
-    private boolean fireRay = true;
     protected Sphere target1, target2;
-
+    protected Box origin;
+    
     @Override
     protected void addMeshAndListeners() {
-              
-        
-        camera.setTranslateZ(-5000);
-        //First person shooter keyboard movement
-        subScene.setOnKeyPressed(ke -> {
-            // add a flag so we can still move the camera
-            if (ke.getCode().equals(KeyCode.CONTROL)) {
-                fireRay = false;
-            }
-            
-        });
-        
-        subScene.setOnKeyReleased(ke -> {
-            // release flag
-            if (ke.getCode().equals(KeyCode.CONTROL)) {
-                fireRay = true;
-            }
+        camera.setTranslateZ(-2000);
+
+        subScene.addEventHandler(KeyEvent.KEY_PRESSED, event -> {
+            if(event.isControlDown()) {
+                double change = 10.0;
+                KeyCode keycode = event.getCode();                
+                //move Origin
+                if(keycode == KeyCode.W) { origin.setTranslateZ(origin.getTranslateZ() + change); }
+                if(keycode == KeyCode.S) { origin.setTranslateZ(origin.getTranslateZ() - change); }
+                if(keycode == KeyCode.A) { origin.setTranslateX(origin.getTranslateX() - change); }
+                if(keycode == KeyCode.D) { origin.setTranslateX(origin.getTranslateX() + change); }
+                if(keycode == KeyCode.SPACE ) { origin.setTranslateZ(origin.getTranslateZ() + change); }
+                if(keycode == KeyCode.Z ) { origin.setTranslateZ(origin.getTranslateZ() - change); }
+            }            
         });
 
-        subScene.setOnMousePressed(e -> {
-            if (fireRay) {
-                Point3D o = CameraHelper.pickProjectPlane(camera, e.getX(), e.getY());
+        subScene.addEventHandler(MouseEvent.MOUSE_PRESSED, e -> {
+            if (e.isControlDown()) {
+                Point3D o = new Point3D(origin.getTranslateX(), origin.getTranslateY(), 
+                    origin.getTranslateZ()); //so it looks like its a bit in front
```

Comment:
s/its/it's

Reply:
oops that comment is no longer applicable. I will remove it.